### DISCUSSION TITLE
next getServerSideProps Function enhancement help

### DISCUSSION THREAD

**Discussion by zq0904** [vercel/next.js#42620] [Help] [answered]
First of all, thank you for this great project, and thank you for your dedication to this project!

The environment
"Next" : "13.0.2",

background
I need to check login information at one of getServerSideProps not login Redirection to login page Users log in and go directly to the page instead of the home page

The sample
```ts
  - pages
    - index.ts // No login is required
    - a.ts // You need to login to access getServerSideProps -> Not logged in redirection login.com? Cb=xxx.com/a so that users login to x.com/automatically jump back

I tried ctx.req.headers.referer or ctx.req.url but it didn't meet my requirements in the following scenarios
  1. When using the Link component at xxx.com, switch to xxx.com/a
    ctx.req.headers.referer // xxx.com
    ctx.req.url // /_next/data/development/a.json
  2. Enter xxx.com in the browser
    ctx.req.headers.referer // undefined
    ctx.req.url // /a 👌 This is in line with my appeal
```

My question
1. Can other fields be provided to support /a
2. How to get protocol (http/https) I can't find any usable fields (maybe Symbol(NextRequestMeta)._protocol but I don't know how)
3. Can you provide a unified api similar to middleware.request for ctx? For example the req. Cookies. The get request. Cookies. Set

**Comment by icyJoseph** [comment, marked as answer]
You can use `resolvedUrl` from the getServerSideProps context object:

```ts
export const getServerSideProps = (ctx: GetServerSidePropsContext) => {
  ctx.resolvedUrl // this one
  return { props: { foo: "bar" } };
};

```

**Reply by zq0904** [reply]
> You can use `resolvedUrl` from the getServerSideProps context object:
> 
> ```ts
> export const getServerSideProps = (ctx: GetServerSidePropsContext) => {
>   ctx.resolvedUrl // this one
>   return { props: { foo: "bar" } };
> };
> ```

Thank you very much. That applies to the first question

**Reply by icyJoseph** [reply]
Yeah, sorry I forgot about the other two.

In general the protocol is a bit tricky because the property that one would traditionally use to check the protocol is not present. If you are behind a proxy, `x-forwarded-proto` is a header you can look for, otherwise, you might want to assume `https://` at all times. Though there might be some questionable implications in doing that.

Not sure about changing the API for cookies though. I imagine you want to have one function that works both in middleware and GSSP, but can't because of the different API's,  however, you can still do type checking to discriminate between them, but it is more cumbersome.

**Reply by zq0904** [reply]
Thank you very much for your answer
